rfc/rfc-92/rfc-92.md

```diff
@@ -0,0 +1,223 @@
+This field will store the serialized and encoded bitmap string directly.
+Each record's key will follow the format: `<column_name>$<column_value>$<file_group_id>`,
+where file_group_id is defined as `<partition_path>$<file_id>`.
+```
+store_type$grocery$20250401$2aa10971-11b2-41b3-9baf-ab33c0000144-0
+```
+
+For non-partitioned tables, we use `.` as a placeholder for the partition path. In this case, the key would be:
+```
+store_type$grocery$.$2aa10971-11b2-41b3-9baf-ab33c0000144-0
+```
+
+The index key can be constructed on-the-fly by concatenating strings during both read and write operations.
+Since the reader must know at least the column name and column value, searching bitmaps by key prefix is also supported.
+
+Below is another example of how the index record looks like in the storage.
+![bitmap_payload](./bitmap_payload.png)
+
+### Maintaining the Index
+Just like other existing Hudi indexes, the logic for writing the bitmap index will be integrated with Hudi's metadata writer during commit operations.
+Bitmap index maintenance during writes typically falls into the following categories: **Initialize**, **Insert**, **Update**, and **Delete**.
+Before diving into each category, let’s define two bitmap meta operations that will be referenced frequently:
+
+**addBitmapPosition**: 
+1. For a given record, extract its indexed column name-value pairs and file group ID to construct the bitmap key.
+2. Retrieve the existing bitmap using the key, or initialize an empty one if it doesn't exist.
+3. Add the record’s position to the bitmap.
+4. Write the bitmap record `<bitmap key, updated bitmap>` to the `bitmap_index` partition in the metadata table
+
+**removeBitmapPosition**:
+1. For a given record, extract its indexed column name-value pairs and file group ID to construct the bitmap key.
+2. Retrieve the existing bitmap using the key; if it doesn’t exist, the operation becomes a no-op.
+3. Remove the record’s position from the bitmap.
+4. Write the bitmap record `<bitmap key, updated bitmap>` to the `bitmap_index` partition in the metadata table
+
+> Note: We don’t need an operation to explicitly delete outdated bitmap records.
+Since the bitmap key remains unchanged, the new record will overwrite the previous one.
+#### Initialize
+When initializing the bitmap index, the metadata writer scans all file groups to collect `existing_records`.
+The bitmap records will be written into the `bitmap_index` partition of the table's Hudi metadata table.
+Pseudocode:
+```
+for record in existing_records:
+  addBitmapPosition(record)
+```
+#### Insert
+The commit metadata for inserts includes files affected by new records.
+The metadata writer extracts `inserted_records` from those files and updates the bitmap index accordingly.
+Pseudocode:
+```
+for record in inserted_records:
+  addBitmapPosition(record)
+```
+#### Update
+Update commit metadata includes files impacted by updated records.
```

Comment:
It lookes like the update strategy is very similiar with RLI, may consider some code/workload resusing from there.

Reply:
There is a refactoring work https://github.com/apache/hudi/pull/12983 going on and I'm planning to use the interface to extract/avoid duplicate code